targets/psa_switch/psa_switch.cpp

```diff
@@ -385,11 +395,22 @@ PsaSwitch::egress_thread(size_t worker_id) {
     Parser *parser = this->get_parser("egress_parser");
     parser->parse(packet.get());
 
+    phv->get_field("psa_egress_input_metadata.egress_port").set(
+        phv->get_field("psa_egress_parser_input_metadata.egress_port"));
```

Comment:
There is no need to change this PR for this purpose (unless you want to add a "TODO" comment to check this later), but while this is correct for unicast packets sent from ingress to egress, there might be other code needed for handling multicast packets, or cloned packets.

Reply:
I think I misplaced this comment.  Copying this value from EgressParser to Egress is always correct, as long as it was correct at the beginning of the EgressParser, which is where it might be useful to have different code for different packet paths.  Not sure yet.

Reply:
I think Andy is right, it seems that "psa_egress_parser_input_metadata.egress_port" is never set. Shouldn't there be a `phv->get_field("psa_egress_parser_input_metadata.egress_port").set(port)` before the egress parser?

Reply:
Right now it seems that we are copying from a "uninitialized" metadata field, which means that its value is likely to be always 0

Reply:
Would it maybe be OK to merge this in, and create an issue to create some STF tests that pass only if all PSA standard metadata fields are being filled in correctly?  Such a P4 PSA program and STF tests wouldn't be quick to write, and should probably be spread over a few different programs rather than only one, but it would be good to have such tests, in any case.  I can write one that only passes if egress_port is filled in correctly, and it should catch any missing cases of initializing this.

Reply:
I'm not sure I understand the rationale for not fixing this PR before merging it. It is a one line fix, unless I am missing something. I'm not asking for a test case here, but since we did bring an issue to light I don't know what would justify not fixing it. After all, the name of the PR is "initialize metadata".

Reply:
The egress_port is enqueued at the end of ingress thread. I thought that would populate the port in egress parser. Did I misunderstand that?

Reply:
I think the _only_ thing that causes a PSA standard metadata field to change is an explicit call to the `set(value)` method on `phv->get_field("psa_egress_parser_input_metadata.egress_port")`.

I have a small STF test I have been hacking on this morning, and can confirm that with the current changes as I write this comment, the egress_port value is 0 while the egress code executes.